implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -37,21 +40,27 @@ default String prefix() {
     }
 
     default String apiToken() {
-        return getSecret(this, "apiToken").required().get();
+        if (apiVersion() == DynatraceApiVersion.V1) {
+            return getSecret(this, "apiToken").required().get();
+        }
+        return getSecret(this, "apiToken").orElse("NO_TOKEN_SET");
     }
 
     default String uri() {
-        return getUrlString(this, "uri").required().get();
+        // for v2, if no URI is set, use the default OneAgent endpoint.
+        // <https://www.dynatrace.com/support/help/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/local-api/>
+        if (apiVersion() == DynatraceApiVersion.V1) {
+            return getUrlString(this, "uri").required().get();
+        }
+        return getUrlString(this, "uri").orElse("http://localhost:14499");
```

Comment:
You could use the full URL here (`http://localhost:14499/metrics/ingest`). Did you verify you're not accidentally appending a `v2` here since that's different from the cluster endpoint?

Reply:
updated the method preparing the endpoint in the exporter v2 and added tests to cover the different cases.